Scripts/Python/xRegisterAge.py

```diff
@@ -0,0 +1,120 @@
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+Additional permissions under GNU GPL version 3 section 7
+
+If you modify this Program, or any covered work, by linking or
+combining it with any of RAD Game Tools Bink SDK, Autodesk 3ds Max SDK,
+NVIDIA PhysX SDK, Microsoft DirectX SDK, OpenSSL library, Independent
+JPEG Group JPEG library, Microsoft Windows Media SDK, or Apple QuickTime SDK
+(or a modified version of those libraries),
+containing parts covered by the terms of the Bink SDK EULA, 3ds Max EULA,
+PhysX SDK EULA, DirectX SDK EULA, OpenSSL and SSLeay licenses, IJG
+JPEG Library README, Windows Media SDK EULA, or QuickTime SDK EULA, the
+licensors of this Program grant you additional
+permission to convey the resulting work. Corresponding Source for a
+non-source form of such a combination shall include the source code for
+the parts of OpenSSL and IJG JPEG Library used as well as that of the covered
+work.
+
+You can contact Cyan Worlds, Inc. by email legal@cyan.com
+ or by snail mail at:
+      Cyan Worlds, Inc.
+      14617 N Newport Hwy
+      Mead, WA   99021
+
+ *==LICENSE==* """
+
+from Plasma import *
+from PlasmaTypes import *
+from PlasmaKITypes import *
+
+act = ptAttribActivator(1, "Act: Buttons")
+resp = ptAttribResponder(2, "Resp: Button Push")
+ageFileName = ptAttribString(3, "age filename")
+ageInstanceName = ptAttribString(4, "age instance name")
+
+class xRegisterAge(ptResponder):
+
+    def __init__(self):
+        # run parent class init
+        ptResponder.__init__(self)
+        self.id = 5008
+        version = 1
+        self.version = version
+        PtDebugPrint("__init__xRegisterAge v.", version)
+        
+    def OnNotify(self,state,id,events):
+        if state and PtFindAvatar(events) == PtGetLocalAvatar():
+            if id==act.id:
+                if not self.IFindAgeLinkInFolder(ageFileName.value):
+                    if resp.value:
+                        resp.run(self.key,events=events)
+                    PtSendKIMessage(kKILocalChatStatusMsg, PtGetLocalizedString("Global.Messages.AgeBookGet", [ageInstanceName.value]))
+                    self.CheckAndRegisterAge(ageFileName.value, ageInstanceName.value)
+
+    def IFindAgeLinkInFolder(self, ageName):
+        vault = ptVault()
+        folder = vault.getAgesIOwnFolder()
+        ageName = ageName.lower()
+        contents = folder.getChildNodeRefList()
+        for content in contents:
+            link = content.getChild().upcastToAgeLinkNode()
+            if link is not None:
+                info = link.getAgeInfo()
+            else:
+                link = content.getChild()
+                info = link.upcastToAgeInfoNode()
+
+            if info.getAgeFilename().lower() == ageName:
+                return link
```

Comment:
Is there a known case where the Age gets stashed into the vault with the wrong case?

Reply:
This function was taken from [nxusBookMachine.py](https://github.com/H-uru/Plasma/blob/af1395d8b6da492817b78224948966238db0c2c3/Scripts/Python/nxusBookMachine.py#L534)
I don't know why this is done this way.